core/src/main/scala/cats/syntax/list.scala

```diff
@@ -1,12 +1,71 @@
 package cats
 package syntax
 
-import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{Ior, IorNel, NonEmptyList, ValidatedNel}
 
 trait ListSyntax {
   implicit def catsSyntaxList[A](la: List[A]): ListOps[A] = new ListOps(la)
 }
 
 final class ListOps[A](val la: List[A]) extends AnyVal {
+
+  /**
+    * Returns an Option of NonEmptyList from a List
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.NonEmptyList
+    * scala> import cats.implicits._
+    *
+    * scala> val result1: List[Int] = List(1, 2)
+    * scala> result1.toNel
+    * res0: Option[NonEmptyList[Int]] = Some(NonEmptyList(1, 2))
+    *
+    * scala> val result2: List[Int] = List.empty[Int]
+    * scala> result2.toNel
+    * res1: Option[NonEmptyList[Int]] = None
+    * }}}
+    */
   def toNel: Option[NonEmptyList[A]] = NonEmptyList.fromList(la)
+
+  /**
+    * Returns a IorNel from a List
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.IorNel
+    * scala> import cats.implicits._
+    *
+    * scala> val result1: List[String] = List("error 1", "error 2")
+    * scala> result1.toIorNel(1)
+    * res0: IorNel[String, Int] = Left(NonEmptyList(error 1, error 2))
+    *
+    * scala> val result2: List[String] = List.empty[String]
+    * scala> result2.toIorNel(1)
+    * res1: IorNel[String, Int] = Right(1)
+    * }}}
+    */
+  def toIorNel[B](ifEmpty: => B): IorNel[A, B] =
```

Comment:
@kailuowang you are right, I didn't make it clear. This is motivated due to a use case I find frequently. If there is some validations made and were returned as a List, meaning that if it is empty everything went well and if it is not then there are errors present, I had to do something like this:

```scala
errorList match {
  case Nil => Valid("this is valid")
  case err :: errs => Invalid(NoneEmptyList(err, errs))
}
```


I found convenient to have a syntax that helped me with this for ValidatedNel and IorNel.

I do agree with you that if there is not consensus on this I can roll it back and maybe try it again on another pull request when I can make an example illustrating the use of what I proposed. I'll wait for your thoughts on this.